src/Processors/QueryPlan/Optimizations/QueryPlanOptimizationSettings.h

```diff
@@ -57,6 +57,7 @@ struct QueryPlanOptimizationSettings
     bool optimize_use_implicit_projections = false;
     bool force_use_projection = false;
     String force_projection_name;
+    size_t max_limit_for_ann_queries;
```

Comment:
Should there be a default initialization value? For example, `0` which ultimately disables the optimization?

Reply:
Good point. It seems weird to me that the other settings above are initialized at all. As far as I see, the code always calls`QueryPlanOptimizationSettings::fromSettings` which makes the implicit initialization pointless.

Let me check if I can refactor this a little bit. I'd do this separately though as people are waiting for this PR :)

Reply:
Needed to merge from `master` anyways and fixed it ...

Reply:
--> https://github.com/ClickHouse/ClickHouse/pull/73693